treebrowser/src/treebrowser.c

```diff
@@ -180,75 +180,35 @@ tree_view_row_expanded_iter(GtkTreeView *tree_view, GtkTreeIter *iter)
 	return expanded;
 }
 
-#if GTK_CHECK_VERSION(3, 10, 0)
-static GdkPixbuf *
-utils_pixbuf_from_name(const gchar *icon_name)
-{
-	GError *error = NULL;
-	GtkIconTheme *icon_theme;
-	GdkPixbuf *pixbuf;
-
-	icon_theme = gtk_icon_theme_get_default ();
-	pixbuf = gtk_icon_theme_load_icon(icon_theme, icon_name, 16, 0, &error);
 
-	if (!pixbuf)
-	{
-		g_warning("Couldn't load icon: %s", error->message);
-		g_error_free(error);
-		return NULL;
-	}
-
-	return pixbuf;
-}
-#else
-static GdkPixbuf *
-utils_pixbuf_from_stock(const gchar *stock_id)
-{
-	GtkIconSet *icon_set;
-
-	icon_set = gtk_icon_factory_lookup_default(stock_id);
-
-	if (icon_set)
-		return gtk_icon_set_render_icon(icon_set, gtk_widget_get_default_style(),
-										gtk_widget_get_default_direction(),
-										GTK_STATE_NORMAL, GTK_ICON_SIZE_MENU, NULL, NULL);
-	return NULL;
-}
-#endif
-
-static GdkPixbuf *
-utils_pixbuf_from_path(gchar *path)
+static GIcon *
+utils_gicon_from_path(gchar *path)
 {
 	GIcon 		*icon;
-	GdkPixbuf 	*ret = NULL;
 	GtkIconInfo *info;
 	gchar 		*ctype;
-	gint 		width;
 
 	ctype = g_content_type_guess(path, NULL, 0, NULL);
 	icon = g_content_type_get_icon(ctype);
 	g_free(ctype);
 
 	if (icon != NULL)
 	{
-		gtk_icon_size_lookup(GTK_ICON_SIZE_MENU, &width, NULL);
-		info = gtk_icon_theme_lookup_by_gicon(gtk_icon_theme_get_default(), icon, width, GTK_ICON_LOOKUP_USE_BUILTIN);
-		g_object_unref(icon);
+		info = gtk_icon_theme_lookup_by_gicon(gtk_icon_theme_get_default(), icon, 16, 0);
```

Comment:
What is usually used for icon sizes? You entered 16 but what about using a gtk constant like before?

Reply:
Notice that the constant GTK_ICON_LOOKUP_USE_BUILTIN  was used for the last parameter which is flags - I believe this is why the icons looked differently on macOS so I used 0.

The width was determined by https://developer.gnome.org/gtk3/stable/gtk3-Themeable-Stock-Images.html#gtk-icon-size-lookup but based on the documentation this function shouldn't be needed and GTK_ICON_SIZE_MENU is 16px. I use the same in ProjectOrganizer and it seems to work fine.

Reply:
Ok, then we leave it as-is. I didn't doubt that it works I just wondered if there is a coding style for stuff like this which says 'use constants or literals.